src/figuresClasses.ts

```diff
@@ -1,19 +1,87 @@
-export interface Figure {
+enum Shapes {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
 }
 
-export class Triangle {
+export interface Figure {
+  shape: Shapes;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape: Shapes = Shapes.Triangle;
 
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (
+      this.a <= 0
+  || this.b <= 0
+  || this.c <= 0
+  || this.a + this.b <= this.c
+  || this.b + this.c <= this.a
+  || this.a + this.c <= this.b
+    ) {
```

Comment:
It's unreadable, for such log conditions - create separates variables or functions

Reply:
Divide it for: <= 0 check and triangle rules check